processing/src/main/java/org/apache/druid/query/groupby/GroupingEngine.java

```diff
@@ -469,13 +465,16 @@ public QueryRunner<ResultRow> mergeRunners(
    * @param query                 the groupBy query
    * @param cursorFactory         cursor factory for the segment in question
    * @param timeBoundaryInspector time boundary inspector for the segment in question
+   * @param bufferPool            processing buffer pool
+   * @param groupByQueryMetrics   metrics instance, will be populated if nonnull
    *
    * @return result sequence for the cursor factory
    */
   public Sequence<ResultRow> process(
       GroupByQuery query,
       CursorFactory cursorFactory,
       @Nullable TimeBoundaryInspector timeBoundaryInspector,
+      NonBlockingPool<ByteBuffer> bufferPool,
```

Comment:
it does feel kind of weird to pass in some resources to the process method since like part of the resources are still associated with the engine via `GroupByResourcesReservationPool`. I'm not entirely sure what is better to do though, so i don't think this is a blocker or anything

Reply:
Could provide the `GroupByResourcesReservationPool` to `process` also, I suppose.